ledger/src/blockstore_processor.rs

```diff
@@ -1395,6 +1395,19 @@ fn load_frozen_forks(
         &mut pending_slots,
     )?;
 
+    // if we started from a validator, we do NOT want to run a final hash calculation
+    let run_final_hash_verification_for_debug = opts
+        .accounts_db_config
+        .as_ref()
+        .map(|config| {
+            config
+                .index
+                .as_ref()
+                .map(|config| !config.started_from_validator)
+                .unwrap_or_default()
```

Comment:
I'm not sure there is a valid case where we'd rely on default from `.unwrap_or_default()`, but default would yield a `false` which would skip the hash_verification.

Should we switch to `.unwrap_or_else(true)`?

Reply:
Validator should always have a config for accounts index as opposed to None. At the moment at least... This is a little brittle. I'm not sure of a better way. We could add a field to the config that says something, like 'debug'? Or 'not-validator' Or, 'run_hash_calc_after_halt_at_slot'.

Reply:
nothing is in the works, but things change over time.

Reply:
We could invert this to say only if the config specifically says we did not start from a validator we run this. Let me see what I can come up with.

Reply:
ok, I changed this usage to be much more narrow and precise with no known future issues possible.